omod/src/main/java/org/openmrs/module/reportingrest/web/controller/ReportingRestController.java

```diff
@@ -28,11 +57,139 @@ public class ReportingRestController extends MainResourceController {
+        if (reportRequest != null) {
+            reportService.purgeReportRequest(reportRequest);
+        }
+    }
+
+    @RequestMapping(value = "/preserveReport", method = RequestMethod.POST)
+    @ResponseStatus(HttpStatus.OK)
+    public void preserveReport(@RequestParam String reportRequestUuid) {
+        ReportService reportService = getReportService();
+        ReportRequest reportRequest = reportService.getReportRequestByUuid(reportRequestUuid);
+        if (ReportRequest.Status.COMPLETED.equals(reportRequest.getStatus())) {
+            Report report = reportService.loadReport(reportRequest);
+            reportService.saveReport(report, StringUtils.EMPTY);
+        }
+    }
+
+    @RequestMapping(value = "/downloadReport", method = RequestMethod.GET)
+    @ResponseStatus(HttpStatus.OK)
+    public FileDownload downloadReport(@RequestParam String reportRequestUuid) {
+        return processAndDownloadReport(reportRequestUuid, getReportService());
+    }
+
+    @RequestMapping(value = "/downloadMultipleReports", method = RequestMethod.GET)
+    @ResponseStatus(HttpStatus.OK)
+    public List<FileDownload> downloadMultipleReports(@RequestParam String reportRequestUuids) {
+        List<FileDownload> fileDownloadList = new ArrayList<FileDownload>();
+        ReportService reportService = getReportService();
+        for (String reportRequestUuid : reportRequestUuids.split(",")) {
+            fileDownloadList.add(processAndDownloadReport(reportRequestUuid, reportService));
+        }
+
+        return fileDownloadList;
+    }
+
+    private FileDownload processAndDownloadReport(String reportRequestUuid, ReportService reportService) {
+        ReportRequest reportRequest = reportService.getReportRequestByUuid(reportRequestUuid);
+        if (reportRequest == null) {
+            throw new IllegalArgumentException("Report request not found");
+        }
+
+        RenderingMode renderingMode = reportRequest.getRenderingMode();
+        if (renderingMode.getRenderer() instanceof WebReportRenderer) {
+            throw new IllegalStateException("Web Renderers not implemented yet");
+        }
+
+        String fileName = renderingMode.getRenderer().getFilename(reportRequest).replace(" ", "_");
+        String contentType = renderingMode.getRenderer().getRenderedContentType(reportRequest);
+        byte[] fileContent = reportService.loadRenderedOutput(reportRequest);
+
+        if (fileContent == null) {
+            throw new IllegalStateException("Error during loading rendered output");
+        } else {
+            return new FileDownload(fileName, contentType, fileContent);
+        }
+    }
+
+    private Object convertParamValueToObject(Object value, Class<?> type) {
+        Object convertedObject = value;
+
+        if (type.equals(Date.class)) {
+            try {
+                convertedObject = DateUtils.parseDate((String) value, "MM/dd/yyyy");
```

Comment:
Why not use `org.openmrs.module.webservices.rest.web.ConversionUtil.convert(value, Date.class)`